.github/workflows/uploadBinaries.yml

```diff
@@ -99,23 +99,42 @@ jobs:
             -DLLVM_FORCE_ENABLE_STATS=${{ matrix.build_config.stats }} \
             -DCIRCT_RELEASE_TAG_ENABLED=ON \
             -DCIRCT_RELEASE_TAG=${{ github.ref_name }} \
-            -DCMAKE_EXPORT_COMPILE_COMMANDS=OFF
+            -DCMAKE_EXPORT_COMPILE_COMMANDS=OFF \
+            -DCMAKE_INSTALL_PREFIX=`pwd`/../install
           ninja
           ninja check-circt check-circt-unit
+          ninja install
+          cd ..
+
       - name: Display Files
-        run: file build/bin/*
+        run: |
+          file install/*
+          file install/bin/*
+
+      - name: Name Install Directory
+        id: name_dir
+        run: |
+          SANITIZED=$(echo ${{ github.ref_name }} | sed 's/\//-/g')
+          echo "value=$SANITIZED" >> "$GITHUB_OUTPUT"
+
       - name: Package Binaries
         run: |
-          cd build
-          ${{ matrix.tar }} czf circt-bin-${{ matrix.runner }}.tar.gz bin
+          mv install ${{ steps.name_dir.outputs.value }}
+          ${{ matrix.tar }} czf circt-bin-${{ matrix.runner }}.tar.gz ${{ steps.name_dir.outputs.value }}
       - name: Show Tarball
         run: |
-          cd build
           ls -l circt-bin-${{ matrix.runner }}.tar.gz
           shasum -a 256 circt-bin-${{ matrix.runner }}.tar.gz
-      - name: Upload Binaries
+      - name: Upload Binaries (Non-Tag)
+        uses: actions/upload-artifact@v3
+        if: github.ref_type != 'tag'
+        with:
+          name: circt-bin-${{ matrix.runner }}.tar.gz
+          path: circt-bin-${{ matrix.runner }}.tar.gz
+          retention-days: 7
```

Comment:
The default is 90 days and I feel like since this is just for quick validation checks, we don't need to eat that much storage.